.github/GITHUB_ACTIONS.md

```diff
@@ -0,0 +1,104 @@
+# GitHub Actions Setup
+
+This repository uses GitHub Actions for continuous integration and Docker image deployment to DockerHub.
+
+## Workflows
+
+### CI (Continuous Integration)
+- **File**: `.github/workflows/ci.yml`
+- **Triggers**: Push to `main` branch, Pull Requests to `main`
+- **Purpose**: Run automated tests using pytest
+- **Python Versions Tested**: 3.10, 3.11
+- **Backend**: Uses `torch-cpu` for faster CI runs
+
+### Docker Build and Push
+- **File**: `.github/workflows/docker.yml`
+- **Triggers**: 
+  - Push to `main` branch
+  - Git tags (e.g., `v1.0.0`)
+  - Manual workflow dispatch
+- **Purpose**: Build and push Docker images to DockerHub
+- **Backends Built**:
+  - `torch-gpu` - PyTorch with CUDA support
+  - `torch-cpu` - PyTorch CPU-only
+  - `tensorflow-gpu` - TensorFlow with CUDA support
+  - `tensorflow-cpu` - TensorFlow CPU-only
+
+## Required GitHub Secrets
+
+To enable Docker image deployment, you need to set up the following secrets in your GitHub repository:
+
+1. **DOCKERHUB_USERNAME**: DockerHub username with push access to the `mskmind` organization
+2. **DOCKERHUB_TOKEN**: A DockerHub access token (not your password)
+
+### Setting up DockerHub Secrets
+
+1. Go to your GitHub repository settings
+2. Navigate to **Settings > Secrets and variables > Actions**
+3. Click **New repository secret**
+4. Add the following secrets:
+   - Name: `DOCKERHUB_USERNAME`, Value: DockerHub username with access to push to the mskmind organization
+   - Name: `DOCKERHUB_TOKEN`, Value: DockerHub access token for the above username
+
+### Creating a DockerHub Access Token
+
+1. Log in to [Docker Hub](https://hub.docker.com/) with a user account that has push access to the `mskmind` organization
+2. Go to **Account Settings > Security**
+3. Click **New Access Token**
+4. Give it a descriptive name (e.g., "GitHub Actions")
+5. Copy the token and save it as the `DOCKERHUB_TOKEN` secret
+
+## Docker Image Tags
+
+Docker images are tagged using the following scheme:
+
+- `latest-<backend>`: Latest build from the main branch (e.g., `latest-torch-gpu`)
+- `main-<backend>`: Latest build from the main branch (same as latest)
+- `v1.0.0-<backend>`: Specific version tag (e.g., `v1.0.0-torch-gpu`)
+- `1.0-<backend>`: Major.minor version (e.g., `1.0-torch-gpu`)
+
+### Example Docker Pull Commands
+
+```bash
+# Pull latest torch-gpu image
+docker pull mskmind/mussel:latest-torch-gpu
```

Comment:
Is the `mskmind` namespace on Dockerhub public, or is it an MSK repository?  I can't seem to see it with either my public or MSK docker accounts.